ckeditor5_sections.module

```diff
@@ -157,6 +157,61 @@ function ckeditor5_sections_attach_entities(DocumentSection $section) {
   }
 }
 
+/**
+ * Implements hook_delivery_cart_referenced_content().
+ */
+function ckeditor5_sections_delivery_cart_referenced_content(Drupal\Core\Entity\EntityInterface $entity) {
+  if ($entity->getEntityTypeId() === "node") {
+
+    // Extract all fields and look for the ones using sections.
+    $fieldsWithSections = [];
+    $fields = $entity->getFieldDefinitions();
+    if (!empty($fields)) {
+      foreach ($fields as $field) {
+        $type = $field->getType();
+        if ($type === "sections") {
+          $fieldsWithSections[$field->getName()] = $field->getName();
+        }
+      }
+    }
+
+    // If we find any using sections extract the data and look for media to add to delivery.
```

Comment:
Could we just look for referenced entities directly in the first loop? Is there a reason to store the fields inside $fieldsWithSections and then loop over them?

Reply:
Combined into one loop